src/utils/database.ts

```diff
@@ -0,0 +1,389 @@
+import Database from "tauri-plugin-sql-api";
+import {
+  Snippet,
+  SnippetFile,
+  SnippetFileInput,
+  SnippetInput,
+  SnippetNote,
+  SnippetTag,
+} from "@/types/snippets";
+
+const getDb = async () => await Database.load("sqlite:devtools.db");
+
+const success = (message: string) => ({ status: 200, message });
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const failure = (message: string, error?: any) => ({ status: 501, message });
+
+// Insertions
+
+const insertSnippet = async (
+  snippet: SnippetInput,
+  files: SnippetFileInput | SnippetFileInput[]
+) => {
+  const db = await getDb();
+  const snippetId = await db
+    .execute("INSERT INTO snippets (name, path) VALUES (?, ?)", [
```

Comment:
Does this handle escape sequences in the code? 
for example will it store following snippet:

```js
const str = "he said \"don't go there\""
```

or a few other test cases, combination of single quotes, double, tripple (python), double escape (escape of escape)